backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/KotlinObjCClassInfoGenerator.kt

```diff
@@ -117,19 +117,21 @@ internal class KotlinObjCClassInfoGenerator(override val context: Context) : Con
             staticData.cStringLiteral(encoding)
     )
 
-    private fun IrClass.generateImpMethodDescs(): List<ObjCMethodDesc> = this.declarations
-            .filterIsInstance<IrSimpleFunction>()
-            .mapNotNull {
-                val annotation =
-                        it.annotations.findAnnotation(context.interopBuiltIns.objCMethodImp.fqNameSafe) ?:
-                                return@mapNotNull null
-
-                ObjCMethodDesc(
-                        annotation.getAnnotationStringValue("selector"),
-                        annotation.getAnnotationStringValue("encoding"),
-                        it.llvmFunction
-                )
-            }
+    private fun IrClass.generateImpMethodDescs(): List<ObjCMethodDesc> {
+        val noDups = mutableMapOf<String, ObjCMethodDesc>()  // remove duplicates [KT-38234]
```

Comment:
Note that the same can probably be done by keeping most of the code here intact, by adding `.distinctBy { it.selector }` at the end.